src/packages/cli/src/cli.ts

```diff
@@ -39,108 +40,158 @@ const isDocker =
 
 const argv = args(detailedVersion, isDocker);
 
-const flavor = argv.flavor;
+if (argv.detach) {
+  // Start Ganache in a child process, and allow it to run in the background.
+  // The only output to stdout should be the PID of the child process.
+  const module = process.argv[1];
+  const args = process.argv.slice(2);
+  args.splice(args.indexOf("--detach"), 1);
+
+  const child = fork(module, args, {
+    stdio: ["ignore", "ignore", "pipe", "ipc"],
+    detached: true
+  });
+
+  // Any messages output to stderr by the child process (before the `ready`
+  // event is emitted) will be streamed to stderr on the parent.
+  child.stderr.pipe(process.stderr);
+
+  child.on("message", message => {
+    if (message === "ready") {
+      console.log(child.pid);
+
+      // Destroy the ReadableStream exposed by the child process, to allow the
+      // parent to exit gracefully.
+      child.stderr.destroy();
+      child.unref();
+      child.disconnect();
+    }
+  });
 
-const cliSettings = argv.server;
+  child.on("error", err => {
+    // This only happens if there's an error starting the child process, not if
+    // the application throws within the child process.
+    console.error(
+      `An error occurred starting Ganache in detached mode: ${err}`
+    );
+    process.exit(1);
```

Comment:
The `process.exit()` forcefully closes the process immediately without letting running timers or promises finish up. While it is unlikely there are any async things happening here, that doesn't mean things won't change in the future.

The reason we use `process.exit(1)` in the failed `server` initialization case elsewhere in this file is because failing to instantiate the `server` could mean there are promises, websockets, etc, that were started in the background, but something has gone horribly wrong and we now have no way to clean up after ourselves (since we don't get a `server` instance we can call `close()` on). We forcibly close the whole process in an attempt to redeem any poor ~life decisions~ programming chooses we've made. Admittedly not the best solution to the problem, but that is the story as to why I justified the use of `process.exit(1)` there.

```suggestion
    process.exitCode = 1;
```